### PR TITLE
[Enhancement] Support SHOW PRIVILEGES statement

### LINKED ISSUE #19398: Cannot create databases or users via UI in DBeaver
If you pick that starrocks is a mysql database, you can connect but can't "do" anything. 

Screenshot 2023-03-11 at 8 43 15 PM

Screenshot 2023-03-11 at 8 45 05 PM

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19398

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 13 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|1|00.00%|[4218]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowPrivilegesStmt.java|0|12|00.00%|[27, 29, 30, 31, 33, 34, 37, 38, 40, 41, 44, 49]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowPrivilegesStmt.java|12|12|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -186,6 +186,7 @@ statement
     | showPluginsStatement
     | showRepositoriesStatement
     | showOpenTableStatement
+    | showPrivilegesStatement
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:189]: better to arrage it by feature, i.e. put it with other authz statements, like "show grants".

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -1207,6 +1208,9 @@ showRepositoriesStatement
 showOpenTableStatement
     : SHOW OPEN TABLES
     ;
+showPrivilegesStatement
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1211]: ditto

**Review by nshangyiming — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:189]: I’m sorting them in alphabetical order, why don’t you wait for time to sort them by feature later?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowPrivilegesStmt.java|12|12|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**